### PR TITLE
Add browser-nativefs article

### PR DESCRIPTION
**Pull request by tomayac**
(Also fixes minor CSS issues)

### Preview https://excalidraw-blog-git-browser-nativefs.excalidraw.now.sh/browser-nativefs/

### CODE REVIEW

File: content/blog/browser-nativefs/index.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+by dealing with the subtleties of progressive enhancement and making your code as simple as possible.
+
+## Acknowledgements
+
+This article was reviewed by [Joe Medley](https://github.com/jpmedley) and
+[Kayce Basques](https://github.com/kaycebasques).
+Thanks to the [contributors to Excalidraw](https://github.com/excalidraw/excalidraw/graphs/contributors)
+for their work on the project and for reviewing my Pull Requests.
+[Hero image](https://unsplash.com/photos/hXrPSgGFpqQ) by
+[Ilya Pavlov](https://unsplash.com/@ilyapavlov) on Unsplash.
```
Comment (tomayac) [inline content/blog/browser-nativefs/index.md:366]:
```suggestion
```

File: src/components/layoutStyles.css [hunk 2]
```diff
@@ -30,12 +35,22 @@ h6 {
...
 }
 
+blockquote {
+  border-left-color: var(--textNormal);
+}
+
+figcaption {
+  text-align: center;
+  font-size: .9rem;
+}
```
Comment (lipis) [inline src/components/layoutStyles.css:52]: ❤️

**Comment by vjeux** [comment]
Looking good to me, thanks for writing this post and doing all this work :)

File: content/blog/browser-nativefs/index.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+I can, for example, start a drawing on my iPhone,
+save it (technically: download it, since Safari does not support the File System Access API)
+to my iPhone Downloads folder, open the file on my desktop (after transferring it from my phone),
+modify the file, and overwrite it with my changes, or even save it as a new file.
+
+<figure>
+  <img src="iphone-original.png" width="300" alt="An Excalidraw drawing on an iPhone.">
+  <figcaption>
+    Starting an Excalidraw drawing on an iPhone where the File System Access API is not supported,
+    but where a file can be saved (downloaded) to the Downloads folder.</a>)
```
Comment (lipis) [inline content/blog/browser-nativefs/index.md:258]:
```suggestion
    but where a file can be saved (downloaded) to the Downloads folder.
```

File: content/blog/browser-nativefs/index.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+For saving a file, traditionally, you are limited to _downloading_ a file,
+which works thanks to the
+[`<a download>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download:~:text=download)
+attribute.
+Given a Blob, you can set the anchor's `href` attribute to a `blob:` URL that you can get from the
+[`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
+method.
+{% Aside 'caution' %}
+To prevent memory leaks, always revoke the URL after the download.
+{% endAside %}
```
Comment (lipis) [inline content/blog/browser-nativefs/index.md:69]: Screenshot 2020-12-08 at 7 52 27 PM

Looks a bit strange.. what is `Aside 'caution'`?

File: content/blog/browser-nativefs/index.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+### Saving (rather: downloading) files
+
+For saving a file, traditionally, you are limited to _downloading_ a file,
+which works thanks to the
+[`<a download>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download:~:text=download)
+attribute.
+Given a Blob, you can set the anchor's `href` attribute to a `blob:` URL that you can get from the
+[`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
+method.
+{% Aside 'caution' %}
```
Comment (tomayac) [inline content/blog/browser-nativefs/index.md:67]:
```suggestion
<blockquote>
```

File: content/blog/browser-nativefs/index.md (hunk 4 above)
Comment (tomayac) [inline content/blog/browser-nativefs/index.md:69]:
```suggestion
</blockquote>
```

Comment (tomayac) [reply content/blog/browser-nativefs/index.md]: Oops, fixed.

File: content/blog/browser-nativefs/index.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+### Saving (rather: downloading) files
+
+For saving a file, traditionally, you are limited to _downloading_ a file,
+which works thanks to the
+[`<a download>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download:~:text=download)
+attribute.
+Given a Blob, you can set the anchor's `href` attribute to a `blob:` URL that you can get from the
+[`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
+method.
+<blockquote>
```
Comment (tomayac) [inline content/blog/browser-nativefs/index.md:67]:
```suggestion

<blockquote>
```

File: content/blog/browser-nativefs/index.md [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+To prevent memory leaks, always revoke the URL after the download.
```
Comment (tomayac) [inline content/blog/browser-nativefs/index.md:68]:
```suggestion
  To prevent memory leaks, always revoke the URL after the download.
```

**Review by lipis — APPROVED** [review]

File: content/blog/browser-nativefs/index.md [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+};
+```
+
+## Introducing browser-nativefs
+
+As perfectly fine as the File System Access API is,
+it's [not yet widely available](https://caniuse.com/#feat=file-system-access-api).
```
Comment (lipis) [inline content/blog/browser-nativefs/index.md:157]: https://caniuse.com/native-filesystem-api

this seems broken: https://caniuse.com/#feat=file-system-access-api

File: content/blog/browser-nativefs/index.md [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+
+<figure>
+  <img src="caniuse.png"
+       alt="Browser support table for the File System Access API. All browsers are marked as 'no support' or 'behind a flag'.">
+  <figcaption>
+    Browser support table for the File System Access API.
+    (<a href="https://caniuse.com/#feat=file-system-access-api">Source</a>)
```
Comment (lipis) [inline content/blog/browser-nativefs/index.md:164]:
```suggestion
    (<a href="https://caniuse.com/native-filesystem-api">Source</a>)
```

File: content/blog/browser-nativefs/index.md (hunk 8 above)
Comment (tomayac) [reply content/blog/browser-nativefs/index.md:157]: Oh, they accepted my PR to change the name, but left the link target intact. Fair. Thanks!

**Comment by tomayac** [comment]
Thanks for the reviews! This should be ready now to go live whenever you want to release it.

File: content/blog/browser-nativefs/index.md [hunk 10]
```diff
@@ -0,0 +1,365 @@
+---
+title: Reading and writing files and directories with the browser-nativefs library
+date: 2020-12-08
```
Comment (lipis) [inline content/blog/browser-nativefs/index.md:3]:
```suggestion
date: 2020-12-09
```

**Pull request merged**